test-validator/src/lib.rs

```diff
@@ -718,14 +708,17 @@ impl TestValidatorGenesis {
         &self,
         socket_addr_space: SocketAddrSpace,
     ) -> (TestValidator, Keypair) {
+        // Using multi_thread to avoid deadlocks when client spawns tasks.
         let mint_keypair = Keypair::new();
         self.start_with_mint_address(mint_keypair.pubkey(), socket_addr_space)
             .inspect(|test_validator| {
-                let runtime = tokio::runtime::Builder::new_current_thread()
+                let runtime = tokio::runtime::Builder::new_multi_thread()
+                    .worker_threads(2)
```

Comment:
Oh cool! Does this mean we don't need to do `worker-threads = 2` in tests with this change?